adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -408,9 +410,10 @@ private void handleConnectionRequestResult(final MqttEndpoint endpoint,
 
         endpoint.closeHandler(v -> close(endpoint, null));
         endpoint.publishHandler(message -> handlePublishedMessage(MqttContext.fromPublishPacket(message, endpoint)));
-
-        endpoint.subscribeHandler(subscribeMsg -> onSubscribe(endpoint, null, subscribeMsg));
-        endpoint.unsubscribeHandler(unsubscribeMsg -> onUnsubscribe(endpoint, null, unsubscribeMsg));
+        final CommandHandler cmdHandler = new CommandHandler(vertx, getConfig());
```

Comment:
cmdHandler should be parameterized

Reply:
Do you mean use type parameter as below?
`CommandHandler<MqttProtocolAdapterProperties> cmdHandler = new CommandHandler(vertx, getConfig());`

Reply:
Inside of `AbstractVertxBasedMqttProtocolAdapter` you should be able to simply re-use its `T` parameter instead of explicitly specifying `MqttProtocolAdapterProperties` ...